### DISCUSSION TITLE
Get all the text content from a page.

### DISCUSSION THREAD

**Discussion by hawkzey** [pnp/pnpcore#1594] [Q&A] [answered]
I am trying to figure out the best way of collecting all textual data from a page, so that i can do some transformation and eventually include it in search.

Is there a way of extracting this data easily?

The documentation is hard to navigate any help would be appreciated.

**Comment by jansenbe** [comment, marked as answer]
@hawkzey : there's no cheap way to get all the text content as a page consists out of text mixed with web parts that each output content based upon their settings. SharePoint has a search API that you can call to discover content
